README.md

```diff
@@ -58,10 +58,70 @@ build/$ ctest . --output-on-failure
 
 The bci.config file is used by our internal validation environment, please just ignore it.
 
+### Build with dilithium support
+
+Dilithium is an optional feature provided by MoCOCrW. To enable the feature,
+replace the CMake invocation with:
+```
+build/$ cmake -DBUILD_TESTING=True -DDILITHIUM_ENABLED=ON ..
+```
+
+Make sure that the adapted version of libdilithium can be found by the linker.
+
+Note, HSM and dilithium support can be enabled independently via CMake arguments.
+
+#### Dilithium Adaptions
+
+It is not possible to take the bare dilithium implementation. The dilithium implementation was
+adapted. A new function for retrieving the public key from a private key was added. This change is
+required to compile MoCOCrW with dilithium support.
+
+There are two PRs created for dilithium. These can be found in dilithium's [github
+repository](https://github.com/pq-crystals/dilithium/).
+
+The first [PR](https://github.com/pq-crystals/dilithium/pull/68) contains the adaptions for
+retreiving the public key from a private key and is **required** in order to compile MoCOCrW with
+dilithium support.
+
+The second [PR](https://github.com/pq-crystals/dilithium/pull/69) improves the CMake file so that
+the static libraries and the header can be installed using CMake. It is recommended to use this PR
+for compiling and installing dilithium.
+
+To get the local copy of libdilithium with above PRs:
+```
+git clone https://github.com/pq-crystals/dilithium.git && cd dilithium
+git reset --hard 3e9b9f1412f6c7435dbeb4e10692ea58f181ee51
+git checkout -b pub-key-extraction
+git pull origin pull/68/head:pub-key-extraction
+git checkout master
+git checkout -b cmake-improvements
+git pull origin pull/69/head:cmake-improvements
+git checkout master
+git merge --no-edit cmake-improvements
+git merge --no-edit pub-key-extraction
+```
+
+and then build with:
+`mkdir build && cd build && cmake .. && cmake --build .`
+
+### Build with HSM support
+
+HSM support is an **optional** feature for MoCOCrW. To build MoCOCrW with HSM support, replace the
```

Comment:
This should probably explain what "HSM" support actually means. Technically, there is support for keys stored in an HSM that offers a PKCS#11 interface.

This should also explain that this feature relies on an adapted version of libp11 as upstream libp11 does not support key creation and does not intend to support it due to the upcoming switch of OpenSSL from the engine API to the provider API.

Reply:
Done